File: companyWise/wiley/chef.java

import java.util.*;


public class chef {
    public static void main(String[] args) {
        Scanner sc= new Scanner(System.in);
        int t= sc.nextInt();
        sc.nextLine();
        String[] data=new String[t];
        for (int i = 0; i < t; i++) {
            data[i]=sc.nextLine();
        }
        int a=Integer.MAX_VALUE;
        int b=Integer.MAX_VALUE;
        int c=Integer.MAX_VALUE;

        for (String ele:data) {
            String[] parts=ele.split(" ");
            switch(parts[0]){
                case "A" -> a=Math.min(a,Integer.parseInt(parts[1]));
                case "B" -> {
                    b=Math.min(b,Integer.parseInt(parts[1]));
                    
                }
                default -> c=Math.min(c,Integer.parseInt(parts[1]));

            }
            
        }
        System.out.println(Math.min(a+b,c));
    }
}

File: companyWise/wiley/chef.py
n=int(input())
arr=[]
for i in range(n):
    arr.append(input().split())

a, b, c = float('inf'), float('inf'), float('inf')
for k in arr:
    if(k[0]=='A'):
        a=min(a, int(k[1]))
    elif(k[0]=='B'):
        b=min(b, int(k[1]))
    else:
        c=min(c, int(k[1]))
print(min(a+b, c))